torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -3158,8 +3158,12 @@ at::Tensor XLANativeFunctions::squeeze_copy(const at::Tensor& self,
 
 at::Tensor XLANativeFunctions::stack(at::TensorList tensors, int64_t dim) {
   TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
+  at::ScalarType result_type = at::native::result_type(tensors);
+  std::vector<at::Tensor> c_tensors(tensors.size());
```

Comment:
is stack expecting input tensor to be CPU? `std::vector<at::Tensor> c_tensors` will return a list of tenosrs on CPU right?

Reply:
No. They are casted tensors, on XLA.

Reply:
Then I am abit confused. Reading your code, you init the `c_tensors` vector which I assume they will be cpu tensors since you didn;t provide the device type. In the later code you only update the dtype of these `c_tensors`, I don't know when are they moved to the XLA device.

Reply:
Here's a summary of what this code is doing: considering the arguments `tensors` (a list of XLA tensors) and `dim`, the function:

1. Computes the common data-type of all tensors: `result_type`
2. Converts each tensor to the common data-type, storing the result in `c_tensors` (as in "cast tensors")
3. Calls `tensor_methods::stack` with the casted tensors

Reply:
Oh I see. `transform` is called with `tensors.begin()`..